tracks/haskell.sh

```diff
@@ -1,10 +1,33 @@
+function _override_stack_resolver() {
+    if [[ -z "$XR_HASKELL_STACK_RESOLVER" ]]; then
+        return
+    fi
+
+    # See https://docs.haskellstack.org/en/stable/pantry/#snapshot-location for resolver syntax.
+    # For the purpose of running exercises it's resonable just to allow lts and nightly.
+    if ! [[ $XR_HASKELL_STACK_RESOLVER =~ '^(lts-[0-9\.]+|nightly-[0-9-]+)$' ]]; then
+        echo Skipped resolver override as $XR_HASKELL_STACK_RESOLVER is not allowed.
+        return
+    fi
+
+    if ! stack --resolver="$XR_HASKELL_STACK_RESOLVER" eval '()' >/dev/null 2>/dev/null; then
```

Comment:
```suggestion
    if ! stack --resolver="$XR_HASKELL_STACK_RESOLVER" eval '()' 2>&1 >/dev/null; then
```

Centralized redirection.